osquery/tables/system/darwin/processes.cpp

```diff
@@ -33,11 +33,40 @@ namespace tables {
 // The maximum number of expected memory regions per process.
 #define MAX_MEMORY_MAPS 512
 
+bool isPidRunning(pid_t pid) {
+  struct kinfo_proc process;
+  size_t proc_buffer_size = sizeof(process);
+
+  const u_int path_length = 4;
+  int path[path_length] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, pid};
+  int sysctl_result =
+      sysctl(path, path_length, &process, &proc_buffer_size, NULL, 0);
+
+  if ((sysctl_result == 0) && (proc_buffer_size != 0)) {
+    return true;
+  }
+
+  return false;
+}
+
 std::set<int> getProcList(const QueryContext &context) {
   std::set<int> pidlist;
+
+  int bufsize = proc_listpids(PROC_ALL_PIDS, 0, nullptr, 0);
+  if (bufsize <= 0) {
+    VLOG(1) << "An error occurred retrieving the process list";
+    return pidlist;
+  }
+
   if (context.constraints.count("pid") > 0 &&
-      context.constraints.at("pid").exists(EQUALS)) {
-    pidlist = context.constraints.at("pid").getAll<int>(EQUALS);
+      context.constraints.at("pid").exists(EQUALS) &&
+      context.constraints.at("pid").getAll<int>(EQUALS).size() <
+          (bufsize / sizeof(pid_t))) {
```

Comment:
Maybe a `auto& constraint = context.constraints.at("pid")` for a more readable `if` statement? Might not make a difference.